### ISSUE TITLE
Confusing documentation on @GDScript vs @GlobalScope

### ISSUE THREAD

**Issue by zeritonius** [godotengine/godot-docs#9157]
Here's a quote from:
https://docs.godotengine.org/en/stable/classes/class_%40gdscript.html
and
https://docs.godotengine.org/en/stable/classes/class_%40globalscope.html#class-globalscope

> `@GDScript`
> Built-in GDScript constants, functions, and annotations.
> Description
> A list of GDScript-specific utility functions and annotations accessible from any script.
> 

> `@GlobalScope`
> Global scope constants and functions.
> 
> Description
> A list of global scope enumerated constants and built-in functions.

It's not clear to me why a built-in function belongs to `@GDScript` or to `@GlobalScope`.
Are GDScript functions global scoped as well?

Thank you.

**Comment by AThousandShips** [comment]
They're specific to GDScript, unlike the global ones, why does this matter? What issue does the cause? Or are you just asking questions?

**Comment by zeritonius** [comment]
>>> They're specific to GDScript, unlike the global ones, why does this matter?
In my opionion, the current wording is confusing.
Here's an example:
https://docs.godotengine.org/en/stable/classes/class_%40globalscope.html#class-globalscope-method-abs
var a = abs(-1)
Is this not GDScript? If not, what is it?
If it's GDScript, why abs belongs to @GlobalScope and not to @GDScript?
It matters to me (and possibly to others) as it's not clear what functions belongs where.

>>> What issue does the cause?
I don't understand this question.

>>> Or are you just asking questions?
I subtmitted this because, in my opinion, the description of @GDScript and @GlobalScope could be improved, to match the clarity of rest of the documentation.

To me, "For the entries related to GDScript which can be accessed in any script see @GDScript." does not answer the question:
"@GDScript a part of @GlobalScope or are two disjunct | complementary sets?"

**Comment by zeritonius** [comment]
> Please put any `@` between backticks, or you'll tag people

I fixed it. Thank you for telling me - I wasn't aware of that.

**Comment by AThousandShips** [comment]
I'm asking what the issue here is? What is the problem caused by not knowing the things you're asking here? It says clearly that the `@GDScript` part is about *GDScript*, and not other things like C# or extensions, the `@GDScript` contains things that are specific to GDScript, like annotations

So my question is: Why is any clarification needed, other than curiosity, what problem are you facing because of this? What issue is caused by not knowing this?

**Comment by zeritonius** [comment]
> It says clearly that the `@GDScript` part is about GDScript, and not other things like C#

Thank you! Your explanation is one step clearer to me, compared to:
"For the entries related to GDScript which can be accessed in any script see `@GDScript`" which doesn't mention C#.
Yes, C# is mentioned in the next paragraph, but it's still not clear to me if `@GDScript` is a subset of `@GlobalScope` or not.

> What is the problem caused by not knowing the things you're asking here?

I saw on your profile that you are a contributor, which means that you have a way more clear aerial view than me and you make some connections which I haven't made yet.
I love Godot documentation for this clarity.
My problem is "confusion". If the confusion of a reader is not a problem, then I didn't submit this in the right place and I apologize.

> It's hard to know what to answer if I don't know why you want to know :)

I appreciate your patience and your willingness to help.
Could you please, answer my previous question, about the abs function? Thank you.

**Comment by AThousandShips** [comment]
> Could you please, answer my previous question, about the abs function?

Because it isn't *specific* to GDScript, it's a global helper method, unlike [`convert`](https://docs.godotengine.org/en/stable/classes/class_%40gdscript.html#class-gdscript-method-convert) which *is* specific to GDScript, and can't be used in C# or extensions, because they work differently

> but it's still not clear to me if `@GDScript` is a subset of `@GlobalScope` or not.

It isn't a subset, if it was everything in `@GDScript` would be in `@GlobalScope`, unless you mean something else by a "subset"

---

The confusion of the reader is absolutely relevant, but I'm not sure what you are confused *about*

The `@GDScript` scope is everything that *only* exists in GDScript, it doesn't really matter *except* if you aren't using GDScript, because then you need to know that these parts aren't available to you, especially the annotations like `@export`, it's separate because of that the parts in that scope are unique to GDScript, and to help organize it better

So the question here is:
What problems are you facing by not knowing why they are separate? Are you unable to write code or use these methods correctly because of it?

If not then it sounds more like "I would like to know why this is this way", which is very valid! But it doesn't necessarily need to be explained in the documentation

---

So, in conclusion, I see some details here we might clarify:
* Clarify even further that the `@GDScript` scope is *exclusive* to GDScript, and the `@GlobalScope` isn't, and that it's only in `GDScript` scripts the former can be used (though I'd say it's redundant to say "GDScript exclusive things to use in GDScript scripts"
* Clarify that in GDScript you can access everything without having to use any prefix

Anything else?

**Comment by AThousandShips** [comment]
> `@GDScript` is actually `@GlobalScopeGDScriptOnly` and `@GlobalScope` is `@GlobalScopeAllLanguagesIncludingGDScript`? I've made up the names.

Not really, `@GlobalScope` is just the name, it doesn't imply anything about being global (it isn't in GDExtension, there these methods are handled by `UtilityFunctions`)

> `@GDScript` is global as well.

It is

> `@GDScript` and `@GlobalScope` have no overlapping functions or anything else.

They do not

> `@GlobalScope` is just a list, which contains, among others, functions that don't start with @?
That is, `@GlobalScope` is not a class, even if it's listed in the classes section of the documentation.

It is a collection of globally accessible methods (in GDScript), and the not starting with `@` isn't relevant, do you mean the annotations listed in `@GDScript`?

> I'm trying to understand how `@GDScript` and `@GlobalScope` works and when to use each.

You don't "use" them in any way really

---

> Where should I ask this question "Why the documentation list the inheritance in reversed order breadcrumbs and how to make it left to right?"
> That is, I have a hard time reading
> Inherits: CanvasItem < Node < Object
> instead:
> Inherits: Object > Node > CanvasItem

That's just personal preference, why is it harder to read it when it's one way when it has arrows?

But that's a separate question and the simple answer is "it just is", it's just a matter of design choices, it is easier to read it the way it is to me

**Comment by zeritonius** [comment]
> That's just personal preference

Well, when the personal preference is set in stone, and I can't change it using the interface, I try to find workarounds. It just occured to me that's easier to make that change just form me. I downloaded the documentation for reading offline and I created a script that reversed the order for every page. I also removed the left menu from every page (another personal preference set in stone). This way, I can easily zoom just the text of the page.

> why is it harder to read it when it's one way when it has arrows?

? spaces has it if even read to you for easier text this Is

You've clarified even some questions I didn't know how to ask ("it isn't in GDExtension, there these methods are handled by UtilityFunctions").

Your explanations were very useful to me and I'm very grateful. You've made my day!

**Comment by AThousandShips** [comment]
> ? spaces has it if even read to you for easier text this Is

Not so much, it's the difference between "A is the parent of B is the parent of C" and "C is the child of B is the child of A", not the reversal of sentences

**Comment by tetrapod00** [comment]
The difference between `@GlobalScope` and `@GDScript` was clarified a bit in https://github.com/godotengine/godot/pull/98438.
I *think* that PR is enough to close this issue as completed, but if it isn't, please comment and this issue can be reopened.
